tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c

```diff
@@ -87,14 +130,118 @@ static void prov_reset(void)
 	ASSERT_OK(bt_mesh_prov_enable(BT_MESH_PROV_ADV));
 }
 
+static bt_mesh_input_action_t gact;
+static uint8_t gsize;
+static int input(bt_mesh_input_action_t act, uint8_t size)
+{
+	/* The test system requests the input OOB data earlier than
+	 * the output OOB is generated. Need to release context here
+	 * to allow output OOB creation. OOB will be inserted later
+	 * after the delay.
+	 */
+	gact = act;
+	gsize = size;
+
+	k_work_reschedule(&oob_timer, K_SECONDS(1));
+
+	return 0;
+}
+
+static void delayed_input(struct k_work *work)
+{
+	char oob_data[16];
+	int size = bs_bc_is_msg_received(*oob_channel_id);
+
+	if (size <= 0) {
+		FAIL("OOB data is not gotten");
+	}
+
+	bs_bc_receive_msg(*oob_channel_id, (uint8_t *)oob_data, size);
+
+	switch (gact) {
+	case BT_MESH_PUSH:
+	case BT_MESH_TWIST:
+	case BT_MESH_ENTER_NUMBER:
+		ASSERT_OK(bt_mesh_input_number(strtoul(oob_data, NULL, 10)));
+		break;
+	case BT_MESH_ENTER_STRING:
+		ASSERT_OK(bt_mesh_input_string(oob_data));
+		break;
+	default:
+		FAIL("Unknown input action %u (size %u) requested!", gact, gsize);
+	}
+}
+
+static void prov_input_complete(void)
+{
+	LOG_INF("Input OOB data completed");
+}
+
+static int output_number(bt_mesh_output_action_t action, uint32_t number);
+static int output_string(const char *str);
+static void capabilities(const struct bt_mesh_dev_capabilities *cap);
 static struct bt_mesh_prov prov = {
 	.uuid = dev_uuid,
 	.unprovisioned_beacon = unprovisioned_beacon,
 	.complete = prov_complete,
 	.node_added = prov_node_added,
+	.output_number = output_number,
+	.output_string = output_string,
+	.input = input,
+	.input_complete = prov_input_complete,
+	.capabilities = capabilities,
 	.reset = prov_reset,
 };
 
+static int output_number(bt_mesh_output_action_t action, uint32_t number)
+{
+	char str[16];
+
+	LOG_INF("OOB Number: %u", number);
+	snprintf(str, sizeof(str), "%d", number);
+
+	bs_bc_send_msg(*oob_channel_id, (uint8_t *)str, strlen(str));
```

Comment:
A sidenote: If you want, you can send directly `number` without converting it to a string (and converting it back on the other side). You should be able to do:
`bs_bc_send_msg(*oob_channel_id, (uint8_t *)number, sizeof(number));`